src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -20,15 +20,8 @@
 GRAFANA_AGENT_METRICS_ENDPOINT = "metrics-endpoint"
 GRAFANA_AGENT_GRAFANA_DASHBOARD = "grafana-dashboards-consumer"
 GRAFANA_AGENT_LOGGING_PROVIDER = "logging-provider"
-# Note(rgildein): The status message comes from the helper function `_update_status` in
-# grafana-agent-k8s, which is used to set the status based on the health of the relation.
-# https://github.com/canonical/grafana-agent-operator/blob/1f2443dedc325f31b2cb02eefe0705afa6ac50e1/src/grafana_agent.py#L464  # noqa E503
-GRAFANA_AGENT_MESSAGE = re.compile(
-    r"Missing "
-    r"(?=.*\['grafana-cloud-config'\]|\['grafana-dashboards-provider'\] for grafana-dashboards-consumer)"
-    r"(?=.*\['grafana-cloud-config'\]|\['logging-consumer'\] for logging-provider)"
-    r"(?=.*\['grafana-cloud-config'\]|\['send-remote-write'\] for metrics-endpoint)"
-)
+GRAFANA_AGENT_API = "localhost:12345/agent/api/v1"
```

Comment:
where is this port coming from ? I am scared if they decide to change it this will just fail. I see few solutions: 
- is there a way to config port for the agent instead of relying on default? 
- if there is no config we can at least put link to code where it is set so we can check if it will change 

WDYT ?

Reply:
The grafana-agent-k8s is not configuring this port, that mean it's comming from upstream as default value. I check that this value is not even in the config file of grafana-agent, so I do not see how we can get this value.

Reply:
It is the upstream's default for `-server.http.address` flag (HTTP server listen:port). Right now in the charm it is not open and not configurable (see https://github.com/canonical/grafana-agent-k8s-operator/issues/308), but it is reachable via `juju ssh` commands, so from its own unit.

For more documentation you can check [this](https://grafana.com/docs/agent/latest/static/configuration/flags/#server).

Reply:
@rgildein it would be worth adding a comment that this path is the default in upstream and that we are reaching it from within the agent's unit, otherwise sending requests to it won't work.